x/mongo/driver/operation.go

```diff
@@ -372,6 +373,16 @@ func (op Operation) Execute(ctx context.Context, scratch []byte) error {
 	resetForRetry := func(err error) {
 		retries--
 		prevErr = err
+
+		// Set the previous indefinite error to be returned in any case where the the error does not have a
+		// NoWritesPerfomed label (the definite case).
+		switch err := err.(type) {
+		case WriteCommandError:
```

Comment:
Can only `WriteCommandError`s have the `NoWritesPerformed` label? If not, we may have to case the error to some interface with a `HasErrorLabel` method.

Reply:
NoWritePerformed [can happen on any RetryableWritesError](https://github.com/10gen/mongo/pull/5550/files#diff-311e49fb621973d50c674e4b8f2a9dcf8899be8aa08681f810d48fc1c22d9241R159). That being said, my understanding of the DRIVERS-2327 and SERVER-66479 is that we do not want to persist/propagate errors with the NoWritesPerformed label (as they are definite). 

Given all that, what this switch does is

1. Check to see if we are dealing with a write error
2. If we are dealing with a write error, is it a NoWritePerfomed error?
    - Yes: this is definite and we don't want to propagate it unless nothing follows it up. If nothing follows it up, it will naturally propagate in the tt switch.
    - No: this is indefinite and we do want to propagate it, so set the `prevIndefiniteError` variable